File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    x=0
    y=0
    for i in sizes:
        if i[0] < i[1]:
            temp=i[0]
            i[0] = i[1]
            i[1] = temp
    for i in sizes:
        x = max(i[0], x)
        y = max(i[1], y)
    answer= x*y
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
class Solution {
    public int solution(int[][] sizes) {
        
        int max_i = 0;
        int max_j = 0;
        
        for (int[] size:sizes) {
            
            if (size[0] >= size[1]) {
                max_i=Math.max(max_i, size[0]);
                max_j=Math.max(max_j, size[1]);
            } else {
                max_i=Math.max(max_i, size[1]);
                max_j=Math.max(max_j, size[0]);
            }
        }
        
        return max_i * max_j;
    }
}